pkg/auth/auth.go

```diff
@@ -1,66 +1,136 @@
 package auth
 
 import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
 	"fmt"
+	"net/url"
 	"regexp"
-	"strings"
 
 	"github.com/xenitab/azdo-proxy/pkg/config"
 )
 
+const tokenLenght = 64
+
 type Authorization struct {
-	Endpoints []Endpoint
+	endpoints map[string]Endpoint
 }
 
 type Endpoint struct {
-	Token   string
-	Regexes []*regexp.Regexp
-}
+	Pat        string
+	Domain     string
+	Schema     string
+	Token      string
+	Namespaces []string
+	SecretName string
 
-// GenerateAuthorization creates regex resources to validate tokens and endpoints
-func GenerateAuthorization(c config.Configuration) (*Authorization, error) {
-	baseApi, err := regexp.Compile(`/` + c.Organization + `/_apis\b`)
-	if err != nil {
-		return nil, err
-	}
+	// metadata used for reverse lookup
+	Organization string
+	Project      string
+	Repository   string
 
-	auth := &Authorization{Endpoints: []Endpoint{}}
-	for _, r := range c.Repositories {
-		git, err := regexp.Compile(`/` + c.Organization + `/` + r.Project + `/_git/` + r.Name + `(/.*)?\b`)
-		if err != nil {
-			return nil, err
-		}
+	regexes []*regexp.Regexp
+}
 
-		api, err := regexp.Compile(`/` + c.Organization + `/` + r.Project + `/_apis/git/repositories/` + r.Name + `(/.*)?\b`)
+// NewAuthorization cretes a new authorization from a give configuration.
+func NewAuthorization(cfg config.Configuration) (Authorization, error) {
```

Comment:
Why should we do that? There is no need for reuse and would make things harder to read.

Reply:
The reason I commented on it was that I had a hard time reading and understanding it. Took me a while. That may be because of my inexperience with Go. 🤷‍♂️